contracts/SwitcheoTokenZIL.scilla

```diff
@@ -192,4 +298,23 @@ transition TransferFrom(from: ByStr20, to: ByStr20, amount: Uint128)
     err = CodeInsufficientAllowance;
     ThrowError err
   end
+end
+
+(* @dev: Burn existing tokens. Only contract self can burn.                      *)
+(* @param burn_account: Address of the token_owner whose balance is to decrease.  *)
+(* @param amount:       Number of tokens to be burned.                            *)
+transition Burn(burn_account: ByStr20, amount: Uint128)
```

Comment:
when will this be used? It seems like `Transfer` transition can cover all the operation?

Reply:
Is it for the case where user transfer to token contract instead of lockproxy?

Reply:
I think this transition is not used at all

Reply:
`Burn` is only called by contract itself:

```
procedure handleBurn(address: ByStr20, amount: Uint128)
  m <- minter;
  is_minter = builtin eq m address;
  match is_minter with
  | True =>
    msg_to_burn = {_tag : "Burn"; _recipient : _this_address; _amount : zero; 
                   burn_account : address; amount : amount};
    msgs = one_msg msg_to_burn;
    send msgs
  | False =>
  end
end
```

A message sent out by this procedure.